### PR TITLE
[INJIMOB-1607]: cwt decode

### PR DESCRIPTION
**Pull request by Alka1703**

### CODE REVIEW

File: kotlin/PixelPass/build.gradle.kts [hunk 1]
```diff
@@ -33,6 +33,12 @@ android {
 }
 
 dependencies {
+    implementation(files("libs/cose-java-1.1.0.jar"))
```
Comment (sree96) [inline kotlin/PixelPass/build.gradle.kts:36]: We don't have this from maven is it?

File: kotlin/PixelPass/src/main/java/io/mosip/pixelpass/cose/KeyUtil.kt [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import java.security.spec.KeySpec
+import java.security.spec.X509EncodedKeySpec
+
+object KeyUtil {
+    private val EDDSA_ALGO: String = "EdDSA"
+
+    @Throws(Exception::class)
+    fun oneKeyFromPublicKey(publicKeyStr: String?): OneKey {
+
+        //Why eddsa sec provider instead of bouncy castle?
```
Comment (sree96) [inline kotlin/PixelPass/src/main/java/io/mosip/pixelpass/cose/KeyUtil.kt:19]: WHY?

**Review by sree96 — CHANGES_REQUESTED** [review]

File: kotlin/PixelPass/src/main/java/io/mosip/pixelpass/cose/KeyUtil.kt (hunk 2 above)
Comment (Alka1703) [reply kotlin/PixelPass/src/main/java/io/mosip/pixelpass/cose/KeyUtil.kt:19]: Seems like the implementation was done in hurry and specific to Philsys. This is what we discussed with Vishwa today.

File: kotlin/PixelPass/build.gradle.kts (hunk 1 above)
Comment (Alka1703) [reply kotlin/PixelPass/build.gradle.kts:36]: The published library version has some issues. Dhanendra shared the jar file he had built from their repository locally

**Review by sree96 — CHANGES_REQUESTED** [review]

**Review by sree96 — APPROVED** [review]

File: kotlin/PixelPass/src/main/java/io/mosip/pixelpass/cose/KeyUtil.kt [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        val strReader = StringReader(publicKeyPem)
+        val pemReader = PemReader(strReader)
+        val pemObject = pemReader.readPemObject()
+        val pubKeyBytes = pemObject.content
+        val pubKeySpec = X509EncodedKeySpec(pubKeyBytes)
+        val keyFactory = KeyFactory.getInstance(algorithm, provider)
+        val publicKeyObj = keyFactory.generatePublic(pubKeySpec)
+        return OneKey(publicKeyObj, null)
+    }
+}
```
Comment (swatigoel) [inline kotlin/PixelPass/src/main/java/io/mosip/pixelpass/cose/KeyUtil.kt:42]: Can't we have single method and differentiate it through is isPhilSysQRData?

**Review by swatigoel — CHANGES_REQUESTED** [review]

File: kotlin/PixelPass/src/main/java/io/mosip/pixelpass/cose/KeyUtil.kt (hunk 3 above)
Comment (Alka1703) [reply kotlin/PixelPass/src/main/java/io/mosip/pixelpass/cose/KeyUtil.kt:42]: Okay. I will refactor

**Review by swatigoel — APPROVED** [review]

**Pull request merged**